### PR TITLE
Dependency check #147

### LINKED ISSUE #147: Dependency check
MainApp should do dependency check (that all required JARs are present) before running application.

### PR DESCRIPTION
**Pull request by HansNewbie**
Fixes #147

### CODE REVIEW

File: src/main/java/address/MainApp.java [hunk 1]
```diff
@@ -49,13 +61,87 @@ public void init() throws Exception {
...
+            System.out.println("Manifest can't be read, not running dependency check");
+            e.printStackTrace();
+            return;
+        }
+
+        Attributes attr = manifest.getMainAttributes();
+        String classPath = attr.getValue("Class-path");
+        List<String> dependencies = new ArrayList<>(Arrays.asList(classPath.split("\\s+")));
+
+        processDependenciesException(dependencies);
```
Comment (HansNewbie) [inline src/main/java/address/MainApp.java]: Will rename the method not to use `exception` as it is not java exception.

File: src/main/java/address/MainApp.java [hunk 2]
```diff
@@ -49,13 +61,87 @@ public void init() throws Exception {
...
+
+        List<String> missingDependencies = dependencies.stream()
```
Comment (yl-coder) [inline src/main/java/address/MainApp.java]: extract method?

File: src/main/java/address/MainApp.java [hunk 3]
```diff
@@ -49,13 +61,87 @@ public void init() throws Exception {
...
+                .filter(dependency -> !FileUtil.isFileExists(dependency)).collect(Collectors.toList());
+
+        if (missingDependencies.isEmpty()) {
+            System.out.println("All dependencies are present");
+        } else {
+            String message = "Missing dependencies:\n";
+            for (String missingDependency : missingDependencies) {
+                message += "- " + missingDependency + "\n";
+            }
+            System.out.println(message.trim());
```
Comment (yl-coder) [inline src/main/java/address/MainApp.java]: extract method for this block?

File: src/main/java/address/MainApp.java (hunk 2 above)
Comment (HansNewbie) [inline src/main/java/address/MainApp.java]: Ok.

File: src/main/java/address/MainApp.java (hunk 3 above)
Comment (HansNewbie) [inline src/main/java/address/MainApp.java]: Noted.

**Pull request merged**